src/services/api/movie-service.js

```diff
@@ -0,0 +1,164 @@
+import HttpBaseClient from "../http-base-client";
+
+const ENDPOINTS = {
+    CREATE: "api/movies/store",
+    INDEX: "api/movies/index",
+    SHOW: "api/movies/show/",
+    INC_NUM_VISITS: "api/movies/visit/",
+    COMMENTS: "api/movies/comments/",
+    CREATE_COMMENT: "api/comments/store",
+    POST_REACTION: "api/reactions/store",
+    FAVORITES_BY_USER: "api/favorites/index/user/",
+    ADD_TO_FAVORITES: "api/favorites/store",
+    REMOVE_FROM_FAVORITES: "api/favorites/destroy",
+    UPDATE_FAVORITE: "api/favorites/update",
+    BEST: "api/movies/best",
+    RELATED: "api/movies/genre/",
+};
+
+class MovieService extends HttpBaseClient {
+    createMovie = async (movieData) => {
+        const { title, description, imageUrl, genre } = movieData;
+
+        const newMovie = this.getApiClient()
+            .post(ENDPOINTS.CREATE, {
+                title,
+                description,
+                imageUrl,
+                genre,
+            })
+            .then((response) => {
+                return response.data;
+            });
+
+        return newMovie;
+    };
+
+    getMovies = async () => {
+        const movies = this.getApiClient()
+            .get(ENDPOINTS.INDEX)
+            .then((response) => response.data);
+
+        return movies;
+    };
+
+    selectMovie = async (id) => {
+        const movie = await this.getApiClient()
+            .get(ENDPOINTS.SHOW + id)
+            .then((response) => response.data);
+
+        localStorage.setItem("selectedMovieId", movie.id);
+        localStorage.setItem("selectedMovieTitle", movie.title);
+        localStorage.setItem("selectedMovieDescription", movie.description);
+        localStorage.setItem("selectedMovieImageUrl", movie.imageUrl);
+        localStorage.setItem("selectedMovieGenre", movie.genre);
+        localStorage.setItem("selectedMovieLikes", movie.likes);
+        localStorage.setItem("selectedMovieDislikes", movie.dislikes);
+
+        return movie;
+    };
+
+    incrementNumVisits = async (id) => {
+        const movie = await this.getApiClient()
+            .post(ENDPOINTS.INC_NUM_VISITS + id)
+            .then((response) => response.data);
+
+        localStorage.setItem("selectedMovieVisits", movie.numVisited);
+
+        return movie;
+    };
+
+    selectComments = async (id) => {
+        const comments = await this.getApiClient()
+            .get(ENDPOINTS.COMMENTS + id)
+            .then((response) => response.data);
```

Comment:
Samim tim sto imas `await` keyword, nema potrebe da pises `then` callback, posto `await` to zapravo radi za tebe. Sa ovim then-om ti samo vracas direktno `data` deo response-a umesto celog response-a, ali nema potreba za tim posto to mozes na primer i u samom `return` iskazu da uradis ili da radis destructuring, kako god.
Na primer:
```
const { data } = await this.getApiClient().get(ENDPOINTS.COMMENTS + id);
return data;
```

Reply:
Gde god to imas, trebalo bi ispeglati.

Reply:
Reseno.